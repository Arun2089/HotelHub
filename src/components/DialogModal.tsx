import { ReactNode, FC } from "react";
import { SvgIconProps } from "@mui/material/SvgIcon";
import {
  Dialog,
  DialogActions,
  DialogTitle,
  DialogContent,
  Typography,
  IconButton,
  Box,
} from "@mui/material";
import CloseIcon from "@mui/icons-material/Close";

type ReusableDialogProps = {
  dialogSize?: "xs" | "sm" | "md" | "lg";
  isOpen: boolean;
  title: string;
  subtitle?: ReactNode;
  handleClose: () => void;
  children: ReactNode;
  actions?: ReactNode;
  TitleIcon?: FC<SvgIconProps>;
  className?: string;
  dialogTitleClassName?: string;
  dialogActionClassName?: string;
  iconButtonClasses?: string;
};

const DialogModal = ({
  dialogSize = "sm",
  isOpen,
  title,
  handleClose,
  subtitle,
  children,
  actions,
  TitleIcon,
  className = "",
  dialogTitleClassName = "",
  dialogActionClassName = "",
  iconButtonClasses = "",
}: ReusableDialogProps) => {
  return (
    <Dialog
      fullWidth
      maxWidth={dialogSize}
      open={isOpen}
      onClose={handleClose}
      PaperProps={{
        className: `rounded-xl ${className}`,
      }}
    >
      <DialogTitle
        className={`padding-b-0 mb-4 bg-primary-main ${dialogTitleClassName}`}
      >
        <Box className="flex-row justiy-content-sp align-items-center">
          <Box className="flex-row align-items-center flex-gap-1">
            {TitleIcon && <TitleIcon fontSize="large" className="text-white" />}
            <Typography
              className="text-lg md:text-2xl text-white inherit"
              variant="h5"
            >
              {title}
            </Typography>
            {subtitle && (
              <Typography className="line-h-date align-self-end">
                {subtitle}
              </Typography>
            )}
          </Box>
          <IconButton className={iconButtonClasses} onClick={handleClose}>
            <CloseIcon className="text-white" />
          </IconButton>
        </Box>
      </DialogTitle>
      <DialogContent>{children}</DialogContent>
      {actions && (
        <DialogActions className={`pb-5 pr-6 gap-2 ${dialogActionClassName}`}>
          {actions}
        </DialogActions>
      )}
    </Dialog>
  );
};

export default DialogModal;
