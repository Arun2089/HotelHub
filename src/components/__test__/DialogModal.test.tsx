import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import DialogModal from "../DialogModal";

const handleClose = jest.fn();
const handleSubmit = jest.fn();

const action = (
  <>
    <button onClick={handleClose}>Cancel</button>
    <button onClick={handleSubmit}>Submit</button>
  </>
);
const title = "Test Dialog";
const subtitle = "Test Subtitle";
const content = "Test Content";

describe("DialogModal", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <DialogModal isOpen={true} title="Test Title" handleClose={handleClose}>
        <div>{content}</div>
      </DialogModal>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders dialog with correct title and subtitle", () => {
    render(
      <DialogModal
        isOpen={true}
        title={title}
        subtitle={subtitle}
        handleClose={handleClose}
      >
        <div>{content}</div>
      </DialogModal>
    );

    const dialogTitle = screen.getByText(title);
    const dialogSubtitle = screen.getByText(subtitle);
    const dialogContent = screen.getByText(content);

    expect(dialogTitle).toBeInTheDocument();
    expect(dialogSubtitle).toBeInTheDocument();
    expect(dialogContent).toBeInTheDocument();
  });

  it("calls handleClose when cancel button or close icon is clicked", () => {
    render(
      <DialogModal
        isOpen={true}
        title={title}
        handleClose={handleClose}
        actions={action}
      >
        <div>{content}</div>
      </DialogModal>
    );

    fireEvent.click(screen.getByText("Cancel"));
    fireEvent.click(screen.getByTestId("close-icon"));

    expect(handleClose).toHaveBeenCalledTimes(2);
  });

  it("calls handleSubmit when submit button is clicked", () => {
    render(
      <DialogModal
        isOpen={true}
        title={title}
        handleClose={handleClose}
        actions={action}
      >
        <div>{content}</div>
      </DialogModal>
    );

    const submitButton = screen.getByText("Submit");
    fireEvent.click(submitButton);

    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
