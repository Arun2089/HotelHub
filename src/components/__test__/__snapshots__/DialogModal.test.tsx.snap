// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogModal snapshot test 1`] = `<DocumentFragment />`;
